### ISSUE TITLE
Correctly propagate 'enabled_features' modifications

### ISSUE THREAD

**Issue by arno-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8001]
Setting `enabled_features.bufferDeviceAddress` to true in `GpuShaderInstrumentor::PreCallRecordCreateDevice`
when adding missing features will modify another validator object, one associated to VkInstance,
and "this" validator is associated to a device. enabled_features is not inherited, and besides
would be reset in `GetEnabledDeviceFeatures`.
The switch from one to the other happens in `state_tracker.cpp`, `ValidationStateTracker::PostCallRecordCreateDevice`
TL;DR is the following type of sanity check is currently invalid, but it would be nice to have

**Comment by spencer-lunarg** [comment]
so I don't get the problem here fully

yes, there are currently two `DeviceFeatures enabled_features`, one for the Instance level and one for Device level, but the Instance Level one should never matter for GPU-AV

I think the only call that we use as an instance level is `PostCallRecordGetPhysicalDeviceProperties` and it doesn't care about the `enabled_features` there

-----

So the flow should be

- We call `GetEnabledDeviceFeatures()` which sets it
- We check if not enabled, and set it to true and enable it
- It is marked as `true` everywhere as expected

**Comment by arno-lunarg** [comment]
I updated the issue
